api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -133,6 +153,18 @@
     return Audiences.COLLECTOR;
   }
 
+  /**
+   * Gets the value of {@code key}.
+   *
+   * @param key the key
+   * @param <T> the type
+   * @return the value
+   * @since 4.8.0
+   */
+  default <T> @Nullable T get(final @NonNull AudienceKey<T> key) {
```

Comment:
This works for anything implementing `Audience` directly, but will fail for any `ForwardingAudience` variants currently. We could add something like `Map<Audience, T> getAll(AudienceKey)` (but how will we handle recursive audiences)?

Reply:
Agreed on it being unnecessary, any time I'd use `getAll` I'd already have the knowledge that it's a ForwardingAudience.
In which case I could just cast and get the audiences and iterate.
Would be less convenient, but it's a simpler solution overall.

Reply:
We can revisit if this is necessary at a later date.